android/filament-android/src/main/java/com/google/android/filament/RenderableManager.java

```diff
@@ -282,6 +282,14 @@ public void setMaterialInstanceAt(@EntityInstance int i, @IntRange(from = 0) int
         nSetMaterialInstanceAt(mNativeObject, i, primitiveIndex, materialInstance.getNativeObject());
     }
 
+    // creates a MaterialInstance Java wrapper object for a particular material instance
+    public @NonNull MaterialInstance getMaterialInstanceAt(@EntityInstance int i,
+            @IntRange(from = 0) int primitiveIndex) {
+        long nativeMatInstance = nGetMaterialInstanceAt(mNativeObject, i, primitiveIndex);
+        long nativeMaterial = nGetMaterialAt(mNativeObject, i, primitiveIndex);
+        return new MaterialInstance(nativeMaterial, nativeMatInstance);
```

Comment:
It's a bit unfortunate that this return a new instance every time. First because of allocations, but also because it makes comparing material instances more difficult (you can't just == them).

Reply:
Agreed  :( .  Mathias and I chatted about this, we thought every other solution wasn't worth the added complexity...open to suggestions.